### PR TITLE
proc: support function call injection on arm64

### PR DESCRIPTION
**Pull request by aarzilli**
Support function call injection on arm64 with go1.19

### CODE REVIEW

**Comment by aarzilli** [comment]
looks like it's working even on darwin/arm64.

**Review by derekparker — APPROVED** [review]
LGTM

**Comment by derekparker** [comment]
This is great! Thanks for the quick implementation.

**Comment by polinasok** [comment]
Wow, that was fast. Just for cross referencing, this updates https://github.com/golang/go/issues/50614

**Comment by tinyfluffs** [comment]
Thanks so much!

File: _fixtures/fncall.go [hunk 1]
```diff
@@ -198,6 +210,7 @@ func main() {
 	var pa2 *astruct
 	var str string = "old string value"
 	longstrs := []string{"very long string 0123456789a0123456789b0123456789c0123456789d0123456789e0123456789f0123456789g012345678h90123456789i0123456789j0123456789"}
+	rast3 := [10]string{"one", "two", "three", "four", "five", "six", "seven", "height", "nine", "ten"}
```
Comment (idan-stripe) [inline _fixtures/fncall.go:213]: -h-eight?

**Comment by juancrescente** [comment]
idk if this is related but I'm not able to run it

```
2022-08-08T16:42:23-03:00 error layer=dap recovered panic: <nil> not an Int
goroutine 84 [running]:
runtime/debug.Stack()
	/usr/local/go/src/runtime/debug/stack.go:24 +0x64
github.com/go-delve/delve/service/dap.(*Session).recoverPanic(0x140002be000, {0x102bd2918, 0x1400572c040})
	/Users/user/go/pkg/mod/github.com/go-delve/delve@v1.9.0/service/dap/server.go:559 +0x48
panic({0x102b0e1a0, 0x140009471c0})
	/usr/local/go/src/runtime/panic.go:884 +0x204
go/constant.Uint64Val({0x0?, 0x0?})
	/usr/local/go/src/go/constant/value.go:519 +0x13c
github.com/go-delve/delve/pkg/proc.(*Variable).parseG(0x102bdbaf8?)
	/Users/user/go/pkg/mod/github.com/go-delve/delve@v1.9.0/pkg/proc/variables.go:900 +0x96c
github.com/go-delve/delve/pkg/proc.GetG({0x102bdbaf8, 0x14003ec4460})
	/Users/user/go/pkg/mod/github.com/go-delve/delve@v1.9.0/pkg/proc/variables.go:259 +0xc8
github.com/go-delve/delve/service/api.ConvertThread({0x102bdbaf8, 0x14003ec4460}, 0x0)
	/Users/user/go/pkg/mod/github.com/go-delve/delve@v1.9.0/service/api/conversions.go:108 +0x118
github.com/go-delve/delve/service/debugger.(*Debugger).state(0x14000240210, 0x0)
	/Users/user/go/pkg/mod/github.com/go-delve/delve@v1.9.0/service/debugger/debugger.go:584 +0x400
github.com/go-delve/delve/service/debugger.(*Debugger).Command(0x14000240210, 0x1400336fa10, 0x14005034a80)
	/Users/user/go/pkg/mod/github.com/go-delve/delve@v1.9.0/service/debugger/debugger.go:1276 +0x1240
github.com/go-delve/delve/service/dap.(*Session).resumeOnce(0x140002be000, {0x1029df210, 0x4}, 0x14003e2a1e0)
	/Users/user/go/pkg/mod/github.com/go-delve/delve@v1.9.0/service/dap/server.go:3443 +0x1d4
github.com/go-delve/delve/service/dap.(*Session).resumeOnceAndCheckStop(0x140002be000, {0x1029df210, 0x4}, 0x1400336fae0?)
	/Users/user/go/pkg/mod/github.com/go-delve/delve@v1.9.0/service/dap/server.go:3585 +0x28
github.com/go-delve/delve/service/dap.glob..func1(0x0?, {0x1029df210?, 0x0?}, 0x0?)
	/Users/user/go/pkg/mod/github.com/go-delve/delve@v1.9.0/service/dap/server.go:3581 +0x20
github.com/go-delve/delve/service/dap.(*Session).runUntilStop(0x140002be000, {0x1029df210, 0x4}, 0x102887280?)
	/Users/user/go/pkg/mod/github.com/go-delve/delve@v1.9.0/service/dap/server.go:3573 +0xb4
github.com/go-delve/delve/service/dap.(*Session).runUntilStopAndNotify(0x140002be000, {0x1029df210, 0x4}, 0x1400020e0d8?)
	/Users/user/go/pkg/mod/github.com/go-delve/delve@v1.9.0/service/dap/server.go:3454 +0x30
github.com/go-delve/delve/service/dap.(*Session).stepUntilStopAndNotify(0x140002be000, {0x1029df210, 0x4}, 0x140031e4050?, {0x0, 0x0}, 0x14003e2a1e0)
	/Users/user/go/pkg/mod/github.com/go-delve/delve@v1.9.0/service/dap/server.go:1904 +0x3ac
github.com/go-delve/delve/service/dap.(*Session).onNextRequest(0x0?, 0x1400572c040, 0x0?)
	/Users/user/go/pkg/mod/github.com/go-delve/delve@v1.9.0/service/dap/server.go:1805 +0x138
github.com/go-delve/delve/service/dap.(*Session).handleRequest.func3()
	/Users/user/go/pkg/mod/github.com/go-delve/delve@v1.9.0/service/dap/server.go:698 +0x68
created by github.com/go-delve/delve/service/dap.(*Session).handleRequest
	/Users/user/go/pkg/mod/github.com/go-delve/delve@v1.9.0/service/dap/server.go:696 +0x12e8
```

**Comment by polinasok** [comment]
@juancrescente How did you get this? We need some description of what you did or even better a test case to reproduce. dap logs would be helpful too. Please file an issue with these detailss. 
The feature in this PR was to run "call SomeFunc(...)" from DEBUG CONSOLE. In the dap server stacktrace that would map to `onEvaluateRequest` and `doCall`. Yours looks like an issue with executing a step. Or was there a call right before that that somehow disturbed the balance to lead to this panic?

**Comment by juancrescente** [comment]
@polinasok see https://github.com/go-delve/delve/pull/3098

**Comment by polinasok** [comment]
Ah, awesome!

**Pull request merged**